lib/postgres.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Library made to simplify accessing / connecting to postgres databases,
+ * and to cleanly handle when the pg module isn't installed.
+ * @author mia-pi-git
+ */
+
+import type {Pool as PostgresPool} from 'pg';
+import type {SQLStatement} from 'sql-template-strings';
+
+const DEFAULT_CONFIG = 'Config' in global ? Config.usepostgres : null;
+
+export class PostgresDatabase {
+	private pool: PostgresPool;
+	connected = false;
+	constructor(config = DEFAULT_CONFIG) {
+		this.pool = config ? new (require('pg').Pool)(config) : null!;
+	}
+	async query(statement: string | SQLStatement, values?: any[]) {
+		if (!this.pool) {
+			return null;
+		}
+		if (!this.connected) {
+			await this.pool.connect();
+			this.connected = true;
+		}
```

Comment:
This doesn't make any sense, this.pool.connect will return a new connection, which you discard. this.pool.query by itself does what you want--make a connection, run a query, and then disconnect. The connected flag is superfluous too

Reply:
Ah, i see.